modulemd/v2/modulemd-module-stream-v2.c

```diff
@@ -106,6 +106,129 @@ modulemd_module_stream_v2_finalize (GObject *object)
+                                  ModulemdModuleStream *self_2)
+{
+  ModulemdModuleStreamV2 *v2_self_1 = NULL;
+  ModulemdModuleStreamV2 *v2_self_2 = NULL;
+
+  g_return_val_if_fail (MODULEMD_IS_MODULE_STREAM_V2 (self_1), FALSE);
+  v2_self_1 = MODULEMD_MODULE_STREAM_V2 (self_1);
+  g_return_val_if_fail (MODULEMD_IS_MODULE_STREAM_V2 (self_2), FALSE);
+  v2_self_2 = MODULEMD_MODULE_STREAM_V2 (self_2);
+
+  if (!MODULEMD_MODULE_STREAM_CLASS (modulemd_module_stream_v2_parent_class)
+         ->equals (self_1, self_2))
+    {
+      return FALSE;
+    }
+
+  /*Check property equality*/
+  if (g_strcmp0 (v2_self_1->community, v2_self_2->community) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v2_self_1->description, v2_self_2->description) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v2_self_1->documentation, v2_self_2->documentation) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v2_self_1->summary, v2_self_2->summary) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v2_self_1->tracker, v2_self_2->tracker) != 0)
+    return FALSE;
+
+  if (!modulemd_buildopts_equals (v2_self_1->buildopts, v2_self_2->buildopts))
+    return FALSE;
+
+  if (!modulemd_hash_table_equals (v2_self_1->rpm_components,
+                                   v2_self_2->rpm_components,
+                                   modulemd_component_equals))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (v2_self_1->module_components,
+                                   v2_self_2->module_components,
+                                   modulemd_component_equals))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->module_licenses,
+                                           v2_self_2->module_licenses))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->content_licenses,
+                                           v2_self_2->content_licenses))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (
+        v2_self_1->profiles, v2_self_2->profiles, modulemd_profile_equals))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->rpm_api,
+                                           v2_self_2->rpm_api))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->rpm_artifacts,
+                                           v2_self_2->rpm_artifacts))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->rpm_filters,
+                                           v2_self_2->rpm_filters))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (v2_self_1->servicelevels,
+                                   v2_self_2->servicelevels,
+                                   modulemd_service_level_equals))
+    {
+      return FALSE;
+    }
+
+  /* 
+  //need an equals function for Modulemd.RpmMapEntry
+  // < string, GHashTable <string, Modulemd.RpmMapEntry> >
```

Comment:
Equality checking for this is a little tricky. How can I pass `modulemd_rpm_map_entry_equals` ?

Reply:
You need to create a wrapper function for the comparison.

Reply:
Hmm, looking at this more, you're right. This is going to be a bit trickier than I was originally thinking. You will probably need to write a helper function here to handle this case, rather than using `modulemd_hash_table_equals()` at the top-level here.